### PR TITLE
Add flag for disabling dependent modules identification in json-render

### LINKED ISSUE #2809: render-json does not work anymore (since 0.51.9)
**Describe the bug**

We upgraded terragrunt and can not run `terragrunt render-json` anymore. It seems that the evaluation phase does not finish (or just takes a very long time). 

It seems that terragrunt jumps all the way up to the _git_-repository root and somehow includes all the environments we have defined.

I tracked it down to the latest working version: 0.51.8. From 0.51.9 it is broken. Diff: [0.51.8 to 0.51.9](https://github.com/gruntwork-io/terragrunt/compare/v0.51.8...v0.51.9).

Our structure is as follows

- /
  - terraform/
    - module1
    - module2
  - terragrunt/
    - dev/
      - common.hcl
      - module1/terragrunt.hcl
      - module2/terragrunt.hcl
    - staging/
      - common.hcl
      - module1/terragrunt.hcl
      - module2/terragrunt.hcl
    - prod/
       - common.hcl
       - module1/terragrunt.hcl
       - module2/terragrunt.hcl 

**To Reproduce**

`terragrunt render-json`

**Expected behavior**

The json should be rendered below a second, it should not jump into other env.

**Versions**
- Terragrunt version: 0.53.6 (since 0.51.9)
- Terraform version: v.1.5.7
- Environment details (Ubuntu 20.04, Windows 10, etc.): MacOS 14.1.1 (23B81)

**Additional context**

One interesting observation: if I rename the `.git` folder in the `/` directory of the project, it works as expected and produces a json file very quickly.

### PR DESCRIPTION
**Pull request by denis256**
## Description

Included changes:
* Added CLI flag to disable dependent modules identification to speed up json rendering
* Added integration tests and docs

Fixes #2809.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Added CLI flag `--terragrunt-json-disable-dependent-modules` to disable identification of root modules during `render-json`

### CODE REVIEW

**Comment by ajax-ryzhyi-r** [comment]
Hi @denis256!
Is there a possibility for this PR to be merged? The degraded performance of the `render-json` subcommand is preventing us from upgrading the terragrunt version, so it would be great to see this flag released 🙃

File: cli/commands/render-json/command.go [hunk 1]
```diff
@@ -24,6 +25,11 @@ func NewFlags(opts *options.TerragruntOptions) cli.Flags {
 			Destination: &opts.RenderJsonWithMetadata,
 			Usage:       "Add metadata to the rendered JSON file.",
 		},
+		&cli.BoolFlag{
+			Name:        FlagNameDisableDependentModules,
+			Destination: &opts.JsonDisableDependentModules,
+			Usage:       "Disable identification of dependent modules rendering json config.",
+		},
```
Comment (levkohimins) [inline cli/commands/render-json/command.go:32]:
```suggestion
		&cli.BoolFlag{
			Name:        FlagNameDisableDependentModules,
			EnvVar:      "TERRAGRUNT_JSON_DISABLE_DEPENDENT_MODULES",
			Destination: &opts.JsonDisableDependentModules,
			Usage:       "Disable identification of dependent modules rendering json config.",
		},
```

**Review by levkohimins — CHANGES_REQUESTED** [review]
Added a small suggestion.

**Review by levkohimins — APPROVED** [review]
LGTM!

**Pull request merged**